src/esp_loader.ts

```diff
@@ -618,7 +618,10 @@ export class ESPLoader extends EventTarget {
         ? Math.round((block.length * uncompressedFilesize) / compressedFilesize)
         : block.length;
       position += flashWriteSize;
-      updateProgress(written, filesize);
+      updateProgress(
+        written,
+        Math.ceil(filesize / flashWriteSize) * flashWriteSize
+      );
```

Comment:
`filesize` is initialized on line 586 and it's what we report to the caller of this function. That value should remain accurate.

It feels weird to adjust the filesize, as that's not actually changing. Shouldn't we just cap the written number of bytes to be equal to the filesize? 

```suggestion
      updateProgress(Math.min(written, filesize), filesize);
```

This way it's never above 100%.

I assume that this discrepancy only happens on the last block?

Reply:
The reason I upped the the filesize is because the block is padded in lines 604-606, so technically the number of bytes to write is greater than filesize, so it's a bit more accurate. However, this is only if the data isn't compressed, so I think your suggested change is fine because it will also fix the issue.